examples/java/unit-test/rest/TwilioRestTest.java

```diff
@@ -628,36 +710,121 @@ public void testShouldQueryParamInRequestwithIncorrectResponseCode() {
+        objectMapper.registerModule(new JavaTimeModule());
+        String testResponse =  "{\"credentials\":[{\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Ahoy\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}, {\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Hello\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}], \"meta\": {\"url\":\"" + url + "\", \"next_page_url\":\"" + url + "?PageSize=5" + "\", \"previous_page_url\":\"" + url + "?PageSize=3" + "\", \"first_page_url\":\"" + url + "?PageSize=1" + "\", \"page_size\":4}}";
+        when(twilioRestClient.request(Mockito.any() )).thenReturn(new Response(testResponse, 200));
+        when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+
+        Aws resource = new AwsFetcher(ACCOUNT_SID).fetch(twilioRestClient);
+
+        assertNotNull(resource);
+    }
+
+    @Test(expected = ApiException.class)
+    public void testShouldGetInValidAPICallResponseAWSFetcher() {
+
+        String url = "https://api.twilio.com/v1/Credentials/AWS";
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        String testResponse =  "{\"credentials\":[{\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Ahoy\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}, {\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Hello\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}], \"meta\": {\"url\":\"" + url + "\", \"next_page_url\":\"" + url + "?PageSize=5" + "\", \"previous_page_url\":\"" + url + "?PageSize=3" + "\", \"first_page_url\":\"" + url + "?PageSize=1" + "\", \"page_size\":4}}";
+        when(twilioRestClient.request(Mockito.any() )).thenReturn(new Response(testResponse, 404));
+        when(twilioRestClient.getObjectMapper()).thenReturn(objectMapper);
+
+        Aws resource = new AwsFetcher(ACCOUNT_SID).fetch(twilioRestClient);
+    }
+
+    @Test(expected = ApiConnectionException.class)
+    public void testShouldGetNullAPICallResponseAWSFetcher() {
+
+        String url = "https://api.twilio.com/v1/Credentials/AWS";
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        String testResponse =  "{\"credentials\":[{\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Ahoy\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}, {\"sid\":\"CR12345678123456781234567812345678\", \"test_string\":\"Hello\", \"test_object\":{\"mms\": true, \"sms\":false, \"voice\": false, \"fax\":true}}], \"meta\": {\"url\":\"" + url + "\", \"next_page_url\":\"" + url + "?PageSize=5" + "\", \"previous_page_url\":\"" + url + "?PageSize=3" + "\", \"first_page_url\":\"" + url + "?PageSize=1" + "\", \"page_size\":4}}";
```

Comment:
testResponse is not Aws json string. Change it here and other places as well.
example: testShouldMakeValidAPICallAwsDeleter
If a test is success try to test with assertEquals wherever is possible.

Reply:
Fixed this as well